File: dcp-solutions/2024-10-25/solution.js
```javascript
function longestPalindromicSubstring(s) {
    const n = s.length;
    let start = 0;
    let maxLen = 1;

    function expandAroundCenter(left, right) {
        while (left >= 0 && right < n && s[left] === s[right]) {
            left--;
            right++;
        }
        return s.slice(left + 1, right);
    }

    for (let i = 0; i < n; i++) {
        const len1 = expandAroundCenter(i, i); // odd length palindrome
        const len2 = expandAroundCenter(i, i + 1); // even length palindrome
        let maxLenOfLen1 = len1.length;
        let maxLenOfLen2 = len2.length;

        if (maxLenOfLen1 > maxLen) {
            start = i - Math.floor((maxLenOfLen1 - 1) / 2);
            maxLen = maxLenOfLen1;
        }
        if (maxLenOfLen2 > maxLen && maxLenOfLen2 % 2 === 0) {
            start = i - Math.floor((maxLenOfLen2 - 2) / 2);
            maxLen = maxLenOfLen2;
        }
    }

    return s.slice(start, start + maxLen);
}
```

File: dcp-solutions/2024-10-25/solution.go
```markdown
package main

import (
    "fmt"
)

func longestPalindromicSubstring(s string) string {
    if len(s) == 0 {
        return ""
    }

    maxLen := 1
    start := 0

    for i := 0; i < len(s); i++ {
        len1 := expandAroundCenter(s, i, i)
        len2 := expandAroundCenter(s, i, i+1)

        len3 := min(len1, len2)
        if len3 > maxLen {
            start = i - (len3 - 1) / 2
            maxLen = len3
        }
    }

    return s[start:start+maxLen]
}

func expandAroundCenter(s string, left, right int) int {
    for left >= 0 && right < len(s) && s[left] == s[right] {
        left--
        right++
    }
    return right - left - 1
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}
```